test/integration/good/code-gen/cpp.expected

```diff
@@ -1215,311 +1217,329 @@ static constexpr std::array<const char*, 427> locations_array__ =
  " (in 'complex_scalar.stan', line 34, column 4 to column 13)",
  " (in 'complex_scalar.stan', line 33, column 45 to line 35, column 3)"};
 
-
-std::complex<double>
-foo(std::ostream* pstream__) {
-  using local_scalar_t__ = double;
-  int current_statement__ = 0; 
-  static constexpr bool propto__ = true;
-  (void) propto__;
-  local_scalar_t__ DUMMY_VAR__(std::numeric_limits<double>::quiet_NaN());
-  (void) DUMMY_VAR__;  // suppress unused var warning
-  try {
-    current_statement__ = 405;
-    return stan::math::to_complex();
-  } catch (const std::exception& e) {
-    stan::lang::rethrow_located(e, locations_array__[current_statement__]);
-  }
-  
-}
-
+struct foo8_functor__ {
+template <typename T0__> stan::promote_args_t<T0__>
+  operator()(const std::vector<std::vector<std::complex<T0__>>>& z,
+             std::ostream* pstream__) const;
+};
+struct foo1_functor__ {
+template <typename T0__> stan::promote_args_t<T0__>
+  operator()(const std::complex<T0__>& z, std::ostream* pstream__) const;
+};
+struct foo4_functor__ {
+std::vector<std::complex<double>>
+  operator()(std::ostream* pstream__) const;
+};
+struct foo6_functor__ {
+template <typename
+  T0__>
+  std::vector<std::complex<stan::promote_args_t<T0__>>>
+  operator()(const T0__& r, std::ostream* pstream__) const;
+};
+struct foo10_functor__ {
+template <typename
+  T0__>
+  std::vector<std::vector<std::complex<stan::promote_args_t<T0__>>>>
+  operator()(const std::vector<std::vector<std::complex<T0__>>>& z,
+             std::ostream* pstream__) const;
+};
+struct foo2_functor__ {
+template <typename T0__>
+  std::complex<stan::promote_args_t<T0__>>
+  operator()(const T0__& r, std::ostream* pstream__) const;
+};
+struct foo9_functor__ {
+template <typename
+  T0__>
+  std::vector<std::vector<std::complex<stan::promote_args_t<T0__>>>>
+  operator()(const T0__& r, std::ostream* pstream__) const;
+};
+struct foo3_functor__ {
+template <typename T0__>
+  std::complex<stan::promote_args_t<T0__>>
+  operator()(const std::complex<T0__>& z, std::ostream* pstream__) const;
+};
+struct foo7_functor__ {
+template <typename
+  T0__>
+  std::vector<std::complex<stan::promote_args_t<T0__>>>
+  operator()(const std::vector<std::complex<T0__>>& z, std::ostream*
+  pstream__) const;
+};
+struct foo5_functor__ {
+template <typename T0__> stan::promote_args_t<T0__>
+  operator()(const std::vector<std::complex<T0__>>& z, std::ostream*
+  pstream__) const;
```

Comment:
I'd prefer if it was pretty printed like the following

```suggestion
template <typename T0__> 
inline stan::promote_args_t<T0__> operator()(
  const std::vector<std::complex<T0__>>& z, std::ostream* pstream__) const;
```

Reply:
( I think these could actually be `static` and `inline` though for static you need to remove the const on the rhs of the signature)